src/storage/FileSystem.cpp

```diff
@@ -56,6 +58,15 @@ int FileSystem::getNewFd()
     return thisFd;
 }
 
+std::string FileSystem::getPathForFd(int fd)
+{
+    if (fileDescriptors.count(fd) == 0) {
+        return "";
+    }
+
+    return fileDescriptors[fd].getPath();
+}
+
```

Comment:
This function helps with giving more meaningful logging during filesystem calls (rather than just printing out the integer handle)